py/dml/c_backend.py

```diff
@@ -491,9 +488,35 @@ def generate_attr_getter(fname, node, port, dimsizes, cprefix, loopvars):
     out('}\n\n', preindent = -1)
     splitting_point()
 
+# dimsizes, loopvars, prefix are relative to port.
+def check_attribute(node, port, prefix):
+    config_param = param_str(node, 'configuration', fallback='none')
+    if config_param == 'none':
+        return
+    if not get_long_doc(node):
+        if (node.objtype in ['attribute', 'connect']
+            and config_param == 'required'):
+            report(WNDOCRA(node, node.logname()))
+        elif node.objtype != 'register':
+            report(WNDOC(node, node.logname()))
+    attrname = get_attr_name(prefix, node)
+    register_attribute(node.site, port, attrname)
+    if port and port.objtype in {'port', 'bank'}:
+        register_attribute(node.site, None, "%s_%s" % (port.name, attrname))
```

Comment:
I think this is technically inaccurate, the conditions for creating proxy attr is more complex; IIRC, attributes in a multidim port array or port in group does not get a proxy. It's a silly bug that we can live with (proxies will vanish eventually), so no need to write a test; however maybe code will look nicer if we extract the proxy precondition logic to a function?

Reply:
Oop, correct. I agree with moving the precondition out; the same condition is checkd within `ConfAttrParentProxyInfoParamExpr.mkexpr`.